dashboards-observability/public/components/synthetics/home_view.tsx

```diff
@@ -127,86 +319,34 @@ export const SyntheticHomeTab = (props: LogHistoryProps) => {
   const pplServiceRequestor = async (
     query: string,
     type: string,
-    result: React.Dispatch<React.SetStateAction<Object[]>>
+    result: React.Dispatch<React.SetStateAction<Array<Record<string, any>>>>
   ) => {
-    console.log(query)
     await props.pplService
-      .fetch({ query: (query), format: 'viz' })
+      .fetch({ query, format: 'viz' })
       .then((res) => {
-        var tempVal = [];
+        const tempVal = [];
         for (const field of Object.keys(res[type])) {
-          tempVal.push(res[type][field])
-        };
-        result(tempVal)
-        console.log(tempVal)
+          tempVal.push(res[type][field]);
+        }
+        result(tempVal);
       })
       .catch((error: Error) => {
-        // setIsError(error.stack);
-        result([])
+        result([]);
         console.error(error);
-      })
-      .finally(() => {
-        // setIsLoading(false);
       });
   };
 
-  // columns for the History table
-  const columns = [
-    {
-      field: "monitorName",
-      name: "Monitor Name",
-      render: (monitor: {} | null | undefined) => (
-        <EuiLink href={`#${monitor}`} target="_blank">
-          {monitor}
-        </EuiLink>
-      ),
-      mobileOptions: {
-        show: false
-      },
-    },
-    {
-      field: "status",
-      name: "Status",
-    },
-    {
-      field: "URL",
-      name: "URL",
-      render: (url: {} | null | undefined) => (
-        <EuiLink href={`${url}`} target="_blank">
-          {url}
-        </EuiLink>
-      ),
-      mobileOptions: {
-        show: false
-      },
-    },
-    {
-      field: "response.status",
-      name: "Status Code",
-    },
-    {
-      field: "startTime",
-      name: "Latest Time Polled",
-      type: "date",
-      render: (date: Date) => {
-        var datet: Date = new Date(date)
-        // needed to convert date into local time, as the index stores times in UTC and this page
-        // incorrectly reads it as local time (while being UTC) so the time will be off by the timezoneoffset
-        var utcdate: Date = new Date(datet.getTime() - (datet.getTimezoneOffset() * 60 * 1000))
-        return utcdate.toString()
-      },
-    },
-  ];
-
   // will pull from index logs every 5 seconds to update at that interval
+  useEffect(() => {
+    updateQueryComponents();
+  }, [historyQuery]);
+
   useEffect(() => {
     const interval = setInterval(() => {
-      console.log("timeq: " + start)
-      pplServiceRequestor(pieQuery + filterVal + timeVal + endPieQuery, "data", setPieChartLogs)
-      pplServiceRequestor(historyQuery + filterVal + timeVal, "jsonData", setHistoryLogs)
-    }, 5000);
+      updateQueryComponents();
+    }, TABLE_REFRESH_INTERVAL_TIME);
     return () => clearInterval(interval);
-  }, [historyQuery, pieQuery, filterVal, timeVal]);
+  });
```

Comment:
without dependency array in use effect i think the interval is created on every render?

Reply:
the render part will be removed and a refresh button will replace this.